tests/regression/diffkabi/diffkabi_test.py

```diff
@@ -0,0 +1,145 @@
+"""
+Regression testing using pytest.
+Individual tests are specified using YAML in the tests/regression/test_specs/
+directory. Each test describes a kernel module, two kernel versions between
+which the module is compared, and a list of compared module parameters. For
+each parameter, pairs of corresponding functions (using the parameter) along
+with the expected analysis results must be provided.
+This script parses the test specification and prepares testing scenarions for
+pytest.
+"""
+
+from diffkemp.llvm_ir.build_llvm import LlvmKernelBuilder
+from diffkemp.semdiff.function_diff import functions_diff, Result
+import glob
+import os
+import shutil
+import pytest
+import yaml
+
+specs_path = "tests/regression/diffkabi/test_specs"
+tasks_path = "tests/regression/diffkabi/kernel_modules"
+
+
+def collect_task_specs():
+    """Collecting and parsing YAML files with test specifications."""
+    result = list()
+    cwd = os.getcwd()
+    os.chdir(specs_path)
+    for spec_file_path in glob.glob("*.yaml"):
+        with open(spec_file_path, "r") as spec_file:
+            spec_yaml = yaml.load(spec_file)
+            if "disabled" in spec_yaml and spec_yaml["disabled"] is True:
+                continue
+            spec_id = os.path.splitext(spec_file_path)[0]
+            result.append((spec_id, spec_yaml))
+    os.chdir(cwd)
+    return result
+
+
+specs = collect_task_specs()
+
+
+class TaskSpec:
+    """
+    Task specification representing one testing scenario (a kernel module with
+    a parameter)
+    """
+    def __init__(self, spec):
+        # Values from the YAML file
+        self.old_kernel = spec["old_kernel"]
+        self.new_kernel = spec["new_kernel"]
+        if "control_flow_only" in spec:
+            self.control_flow_only = spec["control_flow_only"]
+        else:
+            self.control_flow_only = False
+        self.debug = spec["debug"] if "debug" in spec else False
+        self.functions = spec["functions"]
+
+
+def prepare_task(spec):
+    """
+    Prepare testing task (scenario). Build old and new modules and copy
+    created files.
+    """
+    spec.built_functions = dict()
+
+    for function, desc in spec.functions.iteritems():
+        # Find the modules
+        first_builder = LlvmKernelBuilder(spec.old_kernel, None,
+                                          debug=spec.debug)
+        first_builder.source.build_cscope_database()
```

Comment:
This is not needed, build_cscope_database is called by find_srcs_for_function.
Same below.